haproxy/templates/configmap-tls.yaml

```diff
@@ -0,0 +1,28 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: "{{ include "haproxy.fullname" . }}-tls"
+data:
+  haproxy.cfg: |-
+    global
+      log stdout local0
+      maxconn  2048
+      tune.bufsize  400000
+      tune.ssl.default-dh-param  2048
+
+    defaults
+      log global
+      option dontlognull
+      timeout connect 5s
+      timeout client 10s
+      timeout server 10s
+      timeout  check 15s
+      timeout  check 15s
+
+    frontend frontend
+      bind *:{{ .Values.haproxy.frontend.port }}
+      mode tcp
+      default_backend backend
+
+    backend backend
+      server ldap {{ .Values.haproxy.backend }} ssl verify required ca-file {{ .Values.haproxy.certificate.path }}
```

Comment:
If we change the TLS specific values to `haproxy.tls`, this line should be changed accordingly and also include the change for `certificate.path` to `certificatePath`.